packages/astro/src/runtime/server/index.ts

```diff
@@ -182,6 +182,17 @@ function formatList(values: string[]): string {
 
 const rendererAliases = new Map([['solid', 'solid-js']]);
 
+/** @internal Assosciate JSX components with a specific renderer (see /src/vite-plugin-jsx/tag.ts) */
+export function __astro_tag_component__(Component: unknown, rendererName: string) {
+	if (!Component) return;
+	if (typeof Component !== 'function') return;
+	Object.defineProperty(Component, 'astro:renderer', {
```

Comment:
If we were to do it this way, a Symbol would be nicer as it won't show up in `Object.getOwnPropertyNames()` and `Object.getOwnPropertyDescriptors()`

Reply:
Yep, great call!